### PR TITLE
Main Classes

### PR DESCRIPTION
**Pull request by SemenChepak**

### CODE REVIEW

File: app/DB_Connectors/MYSQL.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+from logs.Logger.Loger import DataLogger
+
+
+class MYSQLConnector:
+
+    def __init__(self):
+        self.info = Cred('mysql')
+        self._logger = DataLogger(self)
+
+    def __connect__(self):
```
Comment (yuriimchg) [inline app/DB_Connectors/MYSQL.py]:
```suggestion
    def __connect(self):

```

not a magic method, no need to attach the closing underscores

File: app/DB_Connectors/MYSQL.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+        self._logger.warn('Opening the Connection')
+
+        self._con = pymysql.connect(
+            host=self.info._host,
+            user=self.info._user,
+            password=self.info._password,
+            db=self.info._database,
+            cursorclass=pymysql.cursors.DictCursor
```
Comment (yuriimchg) [inline app/DB_Connectors/MYSQL.py]: is that possible to use the connection string extracted from ini file instead?

File: app/DB_Connectors/MYSQL.py [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+
+class MYSQLConnector:
+
+    def __init__(self):
+        self.info = Cred('mysql')
+        self._logger = DataLogger(self)
+
+    def __connect__(self):
+        self._logger.warn('Opening the Connection')
```
Comment (yuriimchg) [inline app/DB_Connectors/MYSQL.py]:
```suggestion
        self._logger.info('Opening the Connection')

```

File: app/DB_Connectors/POSTGRES.py [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+               f'{self.info._host}/' \
+               f'{self.info._database}'
+
+    def write_to_postgres(self, data_frame, table_name: str):
+        pandas.DataFrame(data_frame). \
+            to_sql(
+            con=self.__engine,
+            name=table_name,
+            if_exists='replace'
+        )
```
Comment (yuriimchg) [inline app/DB_Connectors/POSTGRES.py]: the same comments as for MySQL connector

Comment (yuriimchg) [reply app/DB_Connectors/POSTGRES.py]: _+_ there are common methods for either MySQL and Postgres (fetch, execute, write).

I'd rather move them to some parent class.

File: app/creds/CRED_holder.py [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+
+class Cred:
+    def __init__(self, param):
+        CRED = configparser.ConfigParser()
+        CRED.read(PATH)
+
+        self._host = CRED.get(param, 'host')
+        self._database = CRED.get(param, 'database')
+        self._user = CRED.get(param, 'user')
+        self._password = CRED.get(param, 'password')
```
Comment (yuriimchg) [inline app/creds/CRED_holder.py:14]: you can specify an extra property of this class for the DB connection URI

File: app/data_workers/main_classes/extractor.py [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+        if limit:
+
+            self._logger.info(f'start extracting data key: {Query[key]} limit {limit}')
+
+            return self._db_worker.fetch(f"{Query[key]} limit {limit}")
+        else:
+
+            self._logger.info(f'start extracting data key: {Query[key]}')
+
+            return self._db_worker.fetch(Query[key])
```
Comment (yuriimchg) [inline app/data_workers/main_classes/extractor.py]: please, think of some different way of if / else logic in order to reuse the code

Comment (yuriimchg) [inline app/data_workers/main_classes/extractor.py]: please, think of some different way of if / else logic in order to reuse the code

Comment (yuriimchg) [reply app/data_workers/main_classes/extractor.py]: non if/else approach should apply as well 🙂

File: app/data_workers/main_classes/writer.py [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+    def create_csv(self, data_frame):
+        date = int(time.time())
+
+        self._logger.info(f'Starting creating csv name:<<{date}>>')
+
+        keys = data_frame[0].keys()
+
+        self._logger.info(f'Headers {keys}')
+
+        with open(f'{PATH_OUTPUT}\\csv\\cr_{date}', 'w', newline='') as f:
```
Comment (yuriimchg) [inline app/data_workers/main_classes/writer.py]: https://docs.python.org/3/library/os.html#os.sep 
Please, replace path separators to OS independent

Comment (yuriimchg) [reply app/data_workers/main_classes/writer.py]: here and the other occurences

File: app/script.py [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+            row[key] = value
+        return data_frame
+
+
+class BatiaOfData(DataWriter, DataExtractor, DataTrans):
+    def __init__(self):
+        super(DataWriter, self).__init__()
+        super(DataExtractor, self).__init__()
+        super(DataTrans, self).__init__()
+
```
Comment (yuriimchg) [inline app/script.py]: such a multiple inheritance may cause huge pain during scalling. It would be a better practice to use composition instead

**Review by yuriimchg — CHANGES_REQUESTED** [review]
looks pretty nice to me, liked the structure, logging, execution;

here are the comments (some small changes required):
- modules / folders should be lowercase, while classes are expected to be capitalized (first letter only is uppercased); 
- cache files should not be included;

File: app/data_workers/main_classes/extractor.py [hunk 9]
```diff
@@ -1,26 +1,20 @@
-from DB_Connectors.MYSQL import MYSQLConnector
-from SQL_Query.Queryes import Query
-from logs.Logger.Loger import DataLogger
+from database_connectors.mysql import Mconnector
```
Comment (yuriimchg) [inline app/data_workers/main_classes/extractor.py]: MySQL connector seemed to be a better name

File: app/creds/PATH_holder/path_holder.py [hunk 10]
```diff
@@ -1,3 +1,6 @@
-PATH = 'C:\\Users\\schepak\\Desktop\\oop\\app\\creds\\cred_file\\creds.ini'
-PATH_OUTPUT= 'C:\\Users\\schepak\\Desktop\\oop\\app\\output'
-PATH_LOGS = 'C:\\Users\\schepak\\Desktop\\oop\\app\\logs'
\ No newline at end of file
+import os
+
+PATH = 'C:\\Users\\schepak\\Desktop\\oop\\app\\creds\\cred_file\\creds.ini'.replace(os.sep, '/')
+PATH_OUTPUT_CSV = 'C:\\Users\\schepak\\Desktop\\oop\\app\\output\\csv\\'.replace(os.sep, '/')
+PATH_OUTPUT_JSON = 'C:\\Users\\schepak\\Desktop\\oop\\app\\output\\json\\'.replace(os.sep, '/')
+PATH_LOGS = 'C:\\Users\\schepak\\Desktop\\oop\\app\\logs'.replace(os.sep, '\\')
```
Comment (yuriimchg) [inline app/creds/PATH_holder/path_holder.py:6]: does it work in your machine?

File: app/script.py [hunk 11]
```diff
@@ -29,23 +29,22 @@ def add(self, data_frame, key, value):
...
-        super(DataExtractor, self).__init__()
-        super(DataTrans, self).__init__()
-
+        self.writer = Writer()
+        self.extractor = Extractor()
+        self.trans = Trans()
 
 if __name__ == '__main__':
-    batia = BatiaOfData()
+    batia = Batia()
```
Comment (yuriimchg) [inline app/script.py]: even Batia() is suffering from the naming changes 😁

File: app/creds/PATH_holder/path_holder.py (hunk 10 above)
Comment (SemenChepak) [reply app/creds/PATH_holder/path_holder.py:6]: Yes

**Pull request merged**